moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -1582,75 +1577,100 @@ bool PlanningScene::processAttachedCollisionObjectMsg(const moveit_msgs::Attache
-        if (robot_state_->clearAttachedBody(object.object.id))
+        if (robot_state_->clearAttachedBody(object_to_attach.id_))
           ROS_DEBUG_NAMED(LOGNAME, "The robot state already had an object named '%s' attached to link '%s'. "
                                    "The object was replaced.",
-                          object.object.id.c_str(), object.link_name.c_str());
-        robot_state_->attachBody(object.object.id, shapes, poses, object.touch_links, object.link_name,
-                                 object.detach_posture);
-        ROS_DEBUG_NAMED(LOGNAME, "Attached object '%s' to link '%s'", object.object.id.c_str(),
+                          object_to_attach.id_.c_str(), object.link_name.c_str());
+
+        robot_state_->attachBody(object_to_attach.id_, object_to_attach.shapes_, object_to_attach.shape_poses_,
+                                 object.touch_links, object.link_name, object.detach_posture);
+        ROS_DEBUG_NAMED(LOGNAME, "Attached object '%s' to link '%s'", object_to_attach.id_.c_str(),
                         object.link_name.c_str());
       }
-      else
+      // STEP 3.5: For append operations, only update the attached object and don't look in the world
+      else  // if operation == append
       {
-        const robot_state::AttachedBody* ab = robot_state_->getAttachedBody(object.object.id);
-        shapes.insert(shapes.end(), ab->getShapes().begin(), ab->getShapes().end());
-        poses.insert(poses.end(), ab->getFixedTransforms().begin(), ab->getFixedTransforms().end());
+        const robot_state::AttachedBody* ab = robot_state_->getAttachedBody(object_to_attach.id_);
+        object_to_attach.shapes_.insert(object_to_attach.shapes_.end(), ab->getShapes().begin(), ab->getShapes().end());
+        object_to_attach.shape_poses_.insert(object_to_attach.shape_poses_.end(), ab->getFixedTransforms().begin(),
+                                             ab->getFixedTransforms().end());
         trajectory_msgs::JointTrajectory detach_posture =
             object.detach_posture.joint_names.empty() ? ab->getDetachPosture() : object.detach_posture;
-        const std::set<std::string>& ab_touch_links = ab->getTouchLinks();
-        robot_state_->clearAttachedBody(object.object.id);
+        std::set<std::string> touch_links;
+        
         if (object.touch_links.empty())
-          robot_state_->attachBody(object.object.id, shapes, poses, ab_touch_links, object.link_name, detach_posture);
+        {
+          touch_links = ab->getTouchLinks();
+        }
         else
-          robot_state_->attachBody(object.object.id, shapes, poses, object.touch_links, object.link_name,
-                                   detach_posture);
-        ROS_DEBUG_NAMED(LOGNAME, "Added shapes to object '%s' attached to link '%s'", object.object.id.c_str(),
+        {
+          touch_links = std::set<std::string>(std::make_move_iterator(object.touch_links.begin()),
```

Comment:
I'm not sure about the move operator here. `object` is const and hence shouldn't support move.
However, as the compiler accepts this, looks like move becomes standard copy in this case?

Reply:
It wasn't done like this before, but why not appending touch links as well?

Reply:
1) Tbqh this might have been a clang_tidy addition that I just accepted. I'm not deep on memory access.

2) True, that would make sense for the append operation. Let's make it another PR.

Reply:
It's part of my fork already.